packages/tldraw/src/lib/shapes/shared/useAsset.ts

```diff
@@ -0,0 +1,38 @@
+import { TLAssetId, TLShapeId, useEditor, useValueDebounced } from '@tldraw/editor'
+import { useEffect, useState } from 'react'
+
+/** @internal */
+export function useAsset(shapeId: TLShapeId, assetId: TLAssetId | null) {
+	const editor = useEditor()
+	const [url, setUrl] = useState<string | null>(null)
+	const asset = assetId ? editor.getAsset(assetId) : null
+	const shape = editor.getShape(shapeId)
+
+	const shapeScale =
+		asset && shape && 'w' in shape.props && 'w' in asset.props ? shape.props.w / asset.props.w : 1
```

Comment:
instead of trying to extract the scale from the shape like this (different shapes might scale assets in different ways), how to you feel about having the width / scale / whatever passed into the hook? like `useAsset(assetId, {width: 100})` or something

Reply:
yeah, that's fair - i'm sorta ambivalent about it, will change up.